tools/awslint/lib/rules/construct.ts

```diff
@@ -249,7 +249,10 @@ constructLinter.add({
         fqn = typeRef.mapOfType.fqn;
       }
 
-      e.assert(!(fqn === '@aws-cdk/cdk.Token'), `${e.ctx.propsFqn}.${property.name}`);
```

Comment:
Not sure that's what I would expect... Say we rename cdk.Token to something else. Then basically this linter rule will completely be ignored. I was looking for a way to be defensive around FQN changes given this is very loosely coupled. 


This is a general problem we have with all of the core types we refer to in awslint, so I am wondering if we can create a shared facility that will validate that the core types we want to refer to actually exist in the typesystem:

```ts
// pseudo code:

export class CoreTypes {
  constructor(typesystem: reflect.TypeSystem) {
    if (!typesystem.includesAssembly('@aws-cdk/cdk')) { disable-all-checks })
    for each fqn of CORE_TYPES_FQN {
      if (!typesystem.tryFindFqn(fqn)) {
         throw new Error('cannot find core FQN type ${fqn}');
     }
   }
  }

  public get tokenClass() { return this.typesystem.findClass(CORE_TYPES_FQN.Token); }
  //...
}
```